### DISCUSSION TITLE
Berry and access to spi - (for MCP2515 or 'native' esp can bus)

### DISCUSSION THREAD

**Discussion by NiklausS** [arendst/Tasmota#21260] [Q&A]
Hi everyone,
is it possible to access the spi bus with berry? Or is there a reason why this is not in berry?
Background would be a simple driver for can bus - 

Thank you
 Nik

**Comment by sfromis** [comment]
No Berry integration with SPI. One reason for this is that Tasmota strongly favors I2C when possible, and for SPI devices needing speed, implementing a timing-critical driver in Berry is not a good starting point.

**Reply by sfromis** [reply]
When it comes to the CAN bus, there exists a driver to support this. Not tried using it.
https://github.com/arendst/Tasmota/blob/development/tasmota/tasmota_xsns_sensor/xsns_87_can_sniffer.ino

**Reply by pkkrusty** [reply]
Scripting can use built in Canbus capability of ESP32, according to gemu. Haven't yet tried it myself.

**Comment by NiklausS** [comment]
Thanks for all the answers. 
i've seen the xsns_87 - is also found on the 'M5 bus module' which was my first try, but the driver seems to be dedicated for a project, and not generic. 
I've also found some tips in the scripting direction. But as I'm more on the berry side, i was looking for a solution with berry. 
As most of esp have can bus 'onboard' this would have been the most appealing solution. 
I am fairly new to can-bus so it's more about a proof for now.
Cheers 
 Nik

**Reply by sfromis** [reply]
What you have from the linked driver should also be workable with Berry, by executing any relevant commands and picking up the JSON payloads. No direct integration should be needed, as long as the basics are available.
